libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -543,17 +542,43 @@ void CompilerUtils::convertType(
 		else if (targetTypeCategory == Type::Category::FixedPoint)
 		{
 			solAssert(
-				stackTypeCategory == Type::Category::Integer || 
+				stackTypeCategory == Type::Category::Integer ||
 				stackTypeCategory == Type::Category::RationalNumber ||
 				stackTypeCategory == Type::Category::FixedPoint,
 				"Invalid conversion to FixedMxNType requested."
 			);
-			//shift all integer bits onto the left side of the fixed type
+
 			FixedPointType const& targetFixedPointType = dynamic_cast<FixedPointType const&>(_targetType);
-			if (auto typeOnStack = dynamic_cast<IntegerType const*>(&_typeOnStack))
-				if (targetFixedPointType.numBits() > typeOnStack->numBits())
-					cleanHigherOrderBits(*typeOnStack);
-			solUnimplemented("Not yet implemented - FixedPointType.");
+			if (stackTypeCategory == Type::Category::Integer)
+			{
+				auto typeOnStack = dynamic_cast<IntegerType const*>(&_typeOnStack);
```

Comment:
Please always use `dynamic_cast<IntegerType const&>(...)`, i.e. a reference if you know (or assume) that this is the correct type. The reason is that `dynamic_cast` throws an exception if the type is wrong for a reference but returns a null pointer for a pointer.

Reply:
Oh and then you also have to use `auto const&` instead of `auto`, because just a plain `auto` would result in a non-reference type and thus perform a copy.